cve_bin_tool/data_sources/epss_source.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+import csv
+import gzip
+import logging
+import os
+from datetime import datetime, timedelta
+from io import StringIO
+from pathlib import Path
+
+import aiohttp
+
+from cve_bin_tool.data_sources import DISK_LOCATION_BACKUP, DISK_LOCATION_DEFAULT
+from cve_bin_tool.error_handler import ErrorMode
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class Epss_Source:
+    SOURCE = "Epss"
+    CACHEDIR = DISK_LOCATION_DEFAULT
+    BACKUPCACHEDIR = DISK_LOCATION_BACKUP
+    LOGGER = logging.getLogger().getChild("CVEDB")
+    DATA_SOURCE_LINK = "https://epss.cyentia.com/epss_scores-current.csv.gz"
+
+    def __init__(self, error_mode=ErrorMode.TruncTrace):
+        self.epss_data = None
+        self.error_mode = error_mode
+        self.cachedir = self.CACHEDIR
+        self.backup_cachedir = self.BACKUPCACHEDIR
+        self.epss_path = str(Path(self.cachedir) / "epss")
+        self.file_name = os.path.join(self.epss_path, "epss_scores-current.csv")
+
+    async def update_epss(self):
+        """
+        Updates the EPSS data by downloading and parsing the CSV file.
+        Returns:
+            list: A list of tuples containing the parsed EPSS data.
+                  - CVE ID
+                  - Source (always "EPSS" in this case)
+                  - EPSS score
+                  - EPSS percentile
+        """
+        await self.download_and_parse_epss()
+        return self.epss_data
+
+    async def download_and_parse_epss(self):
+        # Downloads and parses the EPSS data from the CSV file.
+        await self.download_epss_data()
+        self.epss_data = self.parse_epss_data()
+
+    async def download_epss_data(self):
+        # Downloads the EPSS CSV file and saves it to the local filesystem.
+        # The download is only performed if the file is older than 24 hours.
+
+        os.makedirs(self.epss_path, exist_ok=True)
+        # Check if the file exists
+        if os.path.exists(self.file_name):
+            # Get the modification time of the file
+            modified_time = os.path.getmtime(self.file_name)
+            last_modified = datetime.fromtimestamp(modified_time)
+
+            # Calculate the time difference between now and the last modified time
+            time_difference = datetime.now() - last_modified
+
+            # Check if the file is older than 24 hours
+            if time_difference < timedelta(hours=24):
```

Comment:
shouldn't this be  `if time_difference > timedelta(hours=24):` as epss data should be downloaded if file is older than 24 hours?